src/theory/strings/inference_manager.cpp

```diff
@@ -279,6 +279,70 @@ void InferenceManager::markInactive(Node n, ExtReducedId id, bool contextDepend)
 void InferenceManager::processConflict(const InferInfo& ii)
 {
   Assert(!d_state.isInConflict());
+  if (ii.getId() == InferenceId::STRINGS_PREFIX_CONFLICT)
+  {
+    bool isSuf = ii.d_idRev;
+    // The shape of prefix conflicts is P1? ^ P2? ^ (= x y)?
```

Comment:
What are the question marks?

Reply:
It denotes "optional", all 3 conjunctions may or may not exist.